### PR TITLE
Lock down Ruby dependencies

### PR DESCRIPTION
**Pull request by jamesstonehill**
It's a good idea to check the Gemfile.lock into version control to guard against breaking changes in any of the Ruby dependencies and to ensure all developers are working with the same dependencies for debugging purposes.

I've used the latest version of Bundler and ensured we're working with the latest versions of all the dependencies specified.

This means that from now on it's a good idea to periodically run `bundle update` to keep up with the latest versions of the dependencies, although this is not required.

I've also changed some file permissions and updated the travis CI config to cache Ruby dependencies so subsequent builds will be quicker.

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -117,4 +117,4 @@ compress_html:
         - development
         - stage
 
-exclude: ["travis.sh", "Rakefile", "README.md", "Gemfile", "Gemfile.lock", "FAQ.md"]
+exclude: ["travis.sh", "Rakefile", "README.md", "Gemfile", "Gemfile.lock", "FAQ.md", "vendor"]
```
Comment (jamesstonehill) [inline _config.yml]: Had to add this to fix this annoying issue. https://github.com/jekyll/jekyll/issues/2938

**Comment by jamesstonehill** [comment]
@sergiokopplin is this project actively maintained? If not then you might want to archive it to indicate this.

**Comment by sergiokopplin** [comment]
Hey @jamesstonehill , sorry for the delay. Yeah, still active! Let's update this PR and merge it! Can you help me with that? Thanks

**Comment by jamesstonehill** [comment]
@sergiokopplin I've updated the PR to be in line with the latest target branch.

The ruby version bump in the travis config fixes the issue CI you're having with the current target branch (The `&` operator wasn't introduced until ruby 2.3 but in CI you're using version `2.2.x`), but it unfortunately uncoverers the HTML proofer issue, but you're probably fine just mering this anyway.

**Comment by sylvainmetayer** [comment]
Hey @jamesstonehill !

Sorry your PR got closed without notice :confused: 

We now use Github actions to deploy site to github pages, which allow use to use custom gems.

Could you provide your PR against master please ? 

cc @sergiokopplin

**Comment by sergiokopplin** [comment]
Agree!! We should open it again

**Pull request closed without merging**